internal/controllers/addon/status_reporting.go

```diff
@@ -41,33 +40,16 @@ func (r *AddonReconciler) handleOCMAddOnStatusReporting(
 			setLastReportedStatus(addon)
 		}
 	}()
-	currentOCMAddonStatus, err := r.getAddonStatus(ctx, addon.Name)
+	//We've added upsert functionality in the status API so modifying ADO to use POST
+	log.Info("Making the Upsert call ")
+	err = r.postAddonStatus(ctx, addon)
 	if err != nil {
-		ocmErr, ok := err.(ocm.OCMError) //nolint
-		// OCM doesnt yet have the status for this addon.
-		// We go ahead and create it.
-		if ok && ocmErr.StatusCode == http.StatusNotFound {
-			log.Info("reporting addon status for the first time.")
-			err = r.postAddonStatus(ctx, addon)
-		}
-		return
-	}
 
-	if OCMAddOnStatusDifferentFromInClusterAddonStatus(currentOCMAddonStatus, addon) {
-		log.Info("patching addon status.")
-		err = r.patchAddonStatus(ctx, addon)
-		return
+		return err
 	}
 	return nil
```

Comment:
Can we move `setLastReportedStatus(addon)` to this block since doesn't make sense to have a defer block for it now since we only  have a single external api call.